patches/minecraft/net/minecraft/item/SpawnEggItem.java.patch

```diff
@@ -0,0 +1,82 @@
+--- a/net/minecraft/item/SpawnEggItem.java
++++ b/net/minecraft/item/SpawnEggItem.java
+@@ -35,18 +_,28 @@
+ 
+ public class SpawnEggItem extends Item {
+    private static final Map<EntityType<?>, SpawnEggItem> field_195987_b = Maps.newIdentityHashMap();
++   private static final java.util.List<SpawnEggItem> MOD_EGGS = new java.util.ArrayList<>();
+    private final int field_195988_c;
+    private final int field_195989_d;
+-   private final EntityType<?> field_200890_d;
++   private final java.util.function.Supplier<? extends EntityType<?>> field_200890_d;
```

Comment:
Probably should not remove the field (even if its final) and there is no reason to name the supplier field like that.

Reply:
What value should I assign to the original field if I make a new field for the supplier?
The supplier field is called this because I didn't change the mapped name of the field.

Reply:
You probably should assign it the `EntityType` once its available.